src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -3900,8 +3900,11 @@ You should consider suppressing the warning only if you're sure that you don't w
   <data name="ERR_InvalidVersionFormat" xml:space="preserve">
     <value>The specified version string does not conform to the required format - major[.minor[.build[.revision]]]</value>
   </data>
+  <data name="ERR_InvalidVersionFormatDeterministic" xml:space="preserve">
+    <value>The specified version string does not conform to the required format - major[.minor[.build[.revision]]] (without wildcards, due to determinism)</value>
```

Comment:
This message is still really hard to understand. I would prefer something like "The specified version string contains wildcards, which are not compatible with determinism. Either remove wildcards from the version string, or disable determinism for this compilation".

Reply:
The problem is the diagnostic can be reached either if you have wildcards, or you did not follow the format. 
I could probably add a an check for presence of wildcard. I'll try that.